app/src/main/java/de/westnordost/streetcomplete/screens/about/ChangelogFragment.kt

```diff
@@ -75,33 +74,57 @@ class WhatsNewDialog(context: Context, sinceVersion: String) : AlertDialog(conte
 
-    inner class ViewHolder(val binding: RowChangelogBinding) : ListAdapter.ViewHolder<Release>(binding) {
-        override fun onBind(with: Release) {
-            binding.titleLabel.text = with.title
-            binding.descriptionLabel.setHtml(with.description)
-        }
-    }
-}
+private fun Resources.getDimensionInSp(@DimenRes resId: Int) =
+    (getDimension(resId) / displayMetrics.scaledDensity).roundToInt()
 
-data class Release(val title: String, val description: String)
+private fun Resources.getDimensionInDp(@DimenRes resId: Int) =
+    (getDimension(resId) / displayMetrics.density).roundToInt()
 
-private suspend fun readChangelog(resources: Resources): List<Release> = withContext(Dispatchers.IO) {
-    resources.getYamlStringMap(R.raw.changelog).map { Release(it.key, addedLinks(it.value)) }
-}
+private fun WebView.setHtmlFromString(html: String) {
+    val textColor = resources.getHexColor(R.color.text)
+    val linkColor = resources.getHexColor(R.color.accent)
+    val dividerColor = resources.getHexColor(R.color.divider)
 
-private fun addedLinks(description: String): String {
-    return description
-        .replace(Regex("(?<=[\\s(]|^)#(\\d+)")) { matchResult ->
-            val issue = matchResult.groupValues[1]
-            "<a href=\"https://github.com/streetcomplete/StreetComplete/issues/$issue\">#$issue</a>"
-        }
-        .replace(Regex("(?<=[\\s(]|^)@([a-zA-Z\\d-]+)")) { matchResult ->
-            val contributor = matchResult.groupValues[1]
-            "<a href=\"https://github.com/$contributor\">$contributor</a>"
-        }
+    val textSize = resources.getDimensionInSp(androidx.appcompat.R.dimen.abc_text_size_body_1_material)
+    val h2Size = resources.getDimensionInSp(androidx.appcompat.R.dimen.abc_text_size_headline_material)
+    val h3Size = resources.getDimensionInSp(androidx.appcompat.R.dimen.abc_text_size_medium_material)
+    val h4Size = resources.getDimensionInSp(androidx.appcompat.R.dimen.abc_text_size_subhead_material)
+
+    val verticalMargin = resources.getDimensionInDp(R.dimen.activity_vertical_margin)
+    val horizontalMargin = resources.getDimensionInDp(R.dimen.activity_horizontal_margin)
+
+    val style = """
+        <meta name="color-scheme" content="dark light">
+        <style>
+            body {
+                margin: ${verticalMargin}px ${horizontalMargin}px;
+                color: $textColor;
+                font-size: ${textSize}px;
+            }
+
+            :link { color: $linkColor; }
+
+            h2, h3, h4 { font-family: sans-serif-condensed; }
+
+            h2 { font-size: ${h2Size}px; }
+            h3 { font-size: ${h3Size}px; }
+            h4 { font-size: ${h4Size}px; }
+
+            h2:not(:first-child) {
+                border-top: 1px solid $dividerColor;
+                padding-top: 1rem;
+            }
+
+            @media (prefers-color-scheme: dark) {}
+        </style>
+    """.trimIndent()
```

Comment:
Too bad that you have to do this extra work to make this look native. If you are already doing that, I think it may be better to actually create a kind of valid html and put the `html` content in `<body></body>` and the above in `<head></head>`?

Anyway, did you test:
- it works fine both on light and dark mode
- text scales with the system setting as other text in the system does, too
?

Reply:
It doesn't seem to make a difference, but creating valid HTML might be a good idea anyway. I changed that in [`41b89e0`](https://github.com/streetcomplete/StreetComplete/pull/5439/commits/41b89e09177669780f74a8c0ffff9dcabd099ec3).

I tested (on Android 13) that the WebView behaves correctly in both light and dark mode and with various text and display size settings, without restarting the app.